crates/oxc_linter/src/rules/react/no_danger_with_children.rs

```diff
@@ -0,0 +1,357 @@
+        return false;
+    };
+    let is_multi_line = text.value.contains("\n");
+    is_multi_line && is_whitespace(text.value.as_str())
+}
+
+/// Given a JSX element, find the JSXAttributeItem with the given name.
+/// If there are spread props, it will search within those as well.
+fn has_jsx_prop(ctx: &LintContext, node: &AstNode, prop_name: &'static str) -> bool {
+    let AstKind::JSXElement(jsx) = node.kind() else {
+        return false;
+    };
+
+    jsx.opening_element.attributes.iter().any(|attr| match attr {
+        JSXAttributeItem::Attribute(attr) => {
+            let JSXAttributeName::Identifier(ident) = &attr.name else {
+                return false;
+            };
+            ident.name == prop_name
+        }
+        JSXAttributeItem::SpreadAttribute(attr) => {
+            let Some(ident) = attr.argument.get_identifier_reference() else {
+                return false;
+            };
+            does_object_var_have_prop_name(ctx, node, &ident.name.as_str(), prop_name)
+        }
+    })
+}
+
+fn does_object_var_have_prop_name(
+    ctx: &LintContext,
+    node: &AstNode,
+    name: &str,
+    prop_name: &str,
+) -> bool {
+    let Some(symbol) = &find_var_in_scope(ctx, node, name) else {
+        return false;
+    };
+
+    let AstKind::VariableDeclarator(var_decl) = symbol.kind() else {
+        return false;
+    };
+
+    let Some(init) = &var_decl.init else {
+        return false;
+    };
+
+    let Expression::ObjectExpression(obj_expr) = init else {
+        return false;
+    };
+
+    obj_expr.properties.iter().any(|prop| match prop {
+        ObjectPropertyKind::ObjectProperty(obj_prop) => {
+            let Some(key) = obj_prop.key.static_name() else {
+                return false;
+            };
+            key == prop_name
+        }
+        ObjectPropertyKind::SpreadProperty(spread_prop) => {
+            let Some(ident) = spread_prop.argument.get_identifier_reference() else {
+                return false;
+            };
+            // If the next symbol is the same as the current symbol, then there is a cycle,
+            // for example: `const props = {...props}`, so we will stop searching.
+            if let Some(next_symbol) = find_var_in_scope(ctx, node, &ident.name.as_str()) {
+                if next_symbol.id() == symbol.id() {
+                    return false;
+                }
+            }
+
+            does_object_var_have_prop_name(ctx, symbol, &ident.name.as_str(), prop_name)
+        }
+    })
+}
+
+/// Given the name of an identifier, find the variable declaration in the current scope.
+fn find_var_in_scope<'c>(
+    ctx: &'c LintContext,
+    node: &AstNode,
+    name: &str,
+) -> Option<&'c AstNode<'c>> {
+    ctx.scopes()
+        .find_binding(node.scope_id(), name)
+        .map(|symbol_id| ctx.semantic().symbol_declaration(symbol_id))
+}
+
+/// Returns whether a given object has a property with the given name.
+fn is_object_with_prop_name(
```

Comment:
This function shares a decent chunk of code with part of `does_object_var_have_prop_name` but I was struggling to figure out how to refactor it cleanly. Thoughts appreciated 🙏

Reply:
You could attept to resolve the variable, then call this function if you find an object? I.e. starting on like 302.

Reply:
I might need an example here, I'm just having trouble seeing it 😅

Reply:
I'll provide one. My comment is non-blocking, this level of duplication is not problematic.